### ISSUE TITLE
Week 6 Assignment Post - components and slots

### ISSUE THREAD

**Issue by ghost** [susanBuck/e15-spring23#9]
From this week's reading, I found out that you can also create a layout using components and slots. For the Bookmark app, I was able to create a layout and complete the index method using components and slots.

<http://bookmark.jacksuwa.online/books/>
Review the code below:

bookmark/resources/views/components/layout.blade.php
```html
<!doctype html>
<html lang='en'>

<head>
    <title>{{ $title ?? 'Bookmark' }}</title>
    <meta charset='utf-8'>
    <link href='/css/bookmark.css' type='text/css' rel='stylesheet'>
</head>

<body>

    {{ $header }}

    <section>
        {{ $slot }}
    </section>

    <footer>
        &copy; Bookmark, Inc.
    </footer>

</body>

</html>
```
bookmark/resources/views/books/index.blade.php
```html
<x-layout>
    <x-slot:title>
        Books
    </x-slot:title>

    <x-slot:header>
        <a href='/'><img src='/images/bookmark-logo@2x.png' id='logo' alt='Bookmark Logo'></a>
    </x-slot:header>

    <h4>Showing all the books</h4>
</x-layout>
```

Components are similar to blade templates as they are reusable and independent. The {{ $slot }} variable contains the content we want to inject into the component. Components can be also used to create reusable alerts, input, and more for the bookmark app. 

GitHub: <https://github.com/jacksuwa/e15/blob/main/bookmark/resources/views/books/index.blade.php>

**Comment by lmarshman** [comment]
Slots seemed like an interesting topic to me as well, but I'm not sure I fully understand when you would use slot in layouts/views vs @section. It seems like they are both accomplishing the same thing, but I assume there must be a difference or it wouldn't exist.

**Comment by susanBuck** [comment]
@lmarshman - There are some technical differences ([ref](https://stackoverflow.com/questions/44212318/laravel-blade-advantage-of-slot-component-vs-include/62544129#62544129)), but mostly they are accomplishing a lot of the same things. 

The components approach is very similar to design patterns you see in popular front-end JS frameworks like React and Vue, which if I had to guess - is probably part of why it was added. It's a mental model many developers are used to, especially working with complex UIs.
